### ISSUE TITLE
Shared library build exposing Siegfried API

### ISSUE THREAD

**Issue by pm64** [richardlehane/siegfried#97]
I wonder the idea of a shared library (Windows DLL, etc) exposing Siegfried's API has ever been considered?

**Comment by richardlehane** [comment]
Not as yet... 

The golang compiler doesn't currently output DLLs (but may do so in the future)  - so Windows shared libraries not possible (yet).

Since golang 1.5 you can build C shared libraries ... I've never explored this myself, but it may be possible to build a .so file that would be callable from a C program (or higher language that supports C calls e.g. python, lua). If anyone wanted to try this I'd be happy to assist.

**Comment by pm64** [comment]
I wonder if we might revisit this issue? I only follow Go as a casual spectator, but from what I've read, this situation has improved over the last 4 years to where the idea of a Windows DLL should now be within reach. I know there are important considerations beyond this, but just thought I'd check to see if this conversation might be resurrected.

**Comment by richardlehane** [comment]
yes, I think this is possible now - I'll take a look soon, it would be good to resolve this issue!

**Comment by pm64** [comment]
Hi @richardlehane, just a quick note to wish you happy St. Patrick's Day, and perhaps check on the status of this? 😊

**Comment by sevein** [comment]
Hi @richardlehane and @pm64,

FWIW, [pygfried](https://github.com/artefactual-labs/pygfried) can do this for Python devs. I've released [v0.10.0](https://pypi.org/project/pygfried/#files) today, providing wheels (pre-built binary package format for Python) for Windows, macOS, and Linux, or you can build from sources if you happen to have the Go 1.23+ compiler installed locally. pygfried only exposes a very small subset of the API but it's recently learned to return a Python object that resembles `sf -json`.

![Identifying a file with pygfried](https://raw.githubusercontent.com/artefactual-labs/pygfried/refs/heads/main/example.png)

**Comment by richardlehane** [comment]
nice @sevein , thanks for sharing!
